tests/unit/keyinput/test_basekeyparser.py

```diff
@@ -90,6 +92,29 @@ def test_read_config_valid(self):
         assert 'foo' in kp.bindings
         assert 'ctrl+x' in kp.special_bindings
 
+    def test_on_keyconfig_changed(self):
+        """Test the handling of changes in the config."""
+        kp = basekeyparser.BaseKeyParser(0, supports_count=False,
+                                         supports_chains=False)
+        kp.read_config = mock.Mock()
+        kp._modename = None
+
+        # No config set so self._modename is None
+        with pytest.raises(AssertionError) as excinfo:
+            kp.on_keyconfig_changed("normal")
+        assert "on_keyconfig_changed called but no section defined!" in str(
+            excinfo.value)
```

Comment:
I'd prefer to split this like this:

```python
expected_text = "on_keyconfig_changed called but no section defined!"
assert expected_text in str(excinfo.value)
```

But why use `in` and not `==`? Probably a complete match should be easy here, no?